### PR TITLE
Use binary accuracy for binary cross-entropy loss

### PR DESCRIPTION
**Pull request by Joshua-Chin**
Currently, the model will default to categorical accuracy when the loss function is binary cross-entropy. It makes more sense to use binary accuracy instead.

### CODE REVIEW

**Comment by fchollet** [comment]
LGTM

**Comment by DannyWeitekamp** [comment]
Hi this change was made in error, and has invalidated several weeks of results for my project.

Consider any target with 3 elements [0,0,1] and the output [0,0,0]. Binary accuracy computes the mean incorrectly as ~(2/3).
```python
K = np
def binary_accuracy(y_true, y_pred):
    return K.mean(K.equal(y_true, K.round(y_pred)))
trivial = [[0,0,0], [0,0,0], [0,0,0]]
t = [[0,0,1], [0,1,0], [1,0,0]]
binary_accuracy(trivial, o)
```
Output:`0.66666666666666663`

**Comment by DannyWeitekamp** [comment]
In general this accuracy is incorrect because each prediction-target pair is not a proper bernoulli trial. For example in 3 way classification it can yield {0,(1/3), (2/3), 1} whereas a proper bernoulli trial must be in {0,1} only. Therefore the accuracy is improperly defined since it is not necessarily the proportion of correct predictions. Rather it is the mean of some variables that can express degrees of correctness, some of which are absurd as shown in the example above.
